tests/test_segmentation.py

```diff
@@ -38,58 +47,119 @@ def test_save_load_segmentation(tmpdir):
-                        "weight": 4,
+                        "clickable": "false",
+                        "color": "#add8e6",
+                        "fill": "true",
                         "opacity": 0.5,
-                        "fill": True,
-                        "fillColor": "black",
-                        "fillOpacity": 0.1,
-                        "clickable": True,
+                        "stroke": "true",
+                        "weight": 4,
                     }
                 },
+                "type": "Feature",
+            }
+        ],
+        "type": "FeatureCollection",
+    }
+    test_seg = Segmentation(test_json)
+
+    test_seg_swapped = swap_coordinates(test_seg)
+    assert test_seg_swapped["features"][0]["geometry"]["coordinates"] == [
+        [[1, 0], [1, 0], [1, 0]]
+    ]
+
+
+def test_get_min_max_values():
+    test_json = {
+        "features": [
+            {
                 "geometry": {
+                    "coordinates": [[[0, 3], [1, 1], [4, 1]]],
                     "type": "Polygon",
-                    "coordinates": [
-                        [
-                            [171.715394, -41.792157],
-                            [171.710101, -41.802235],
-                            [171.728054, -41.800635],
-                            [171.715394, -41.792157],
-                        ]
-                    ],
                 },
+                "properties": {
+                    "style": {
+                        "clickable": "false",
+                        "color": "#add8e6",
+                        "fill": "true",
+                        "opacity": 0.5,
+                        "stroke": "true",
+                        "weight": 4,
+                    }
+                },
+                "type": "Feature",
             }
-        ]
-    )
-    test_map = InteractiveMap(dataset)
+        ],
+        "type": "FeatureCollection",
+    }
+    test_seg = Segmentation(test_json)
+    assert get_min_max_values(test_seg) == {"minX": 0, "maxX": 4, "minY": 1, "maxY": 3}
+
+
+def test_show_map(dataset, boundary_segmentation):
+    # simple test to verify maps can be opened.
+
+    test_map = Map(dataset)
     test_map.show()
-    test_map = InteractiveMap(segmentation=segmentation_2)
+    # todo setup test with segmentations.
+
+    test_map = Map(segmentation=boundary_segmentation)
     test_map.show()
-    # test segmentation as datatype
-    with pytest.raises(TypeError):
-        test_map = InteractiveMap(segmentation_2)
 
+    # test show of segmentation
+    boundary_segmentation.show()
+
+    with pytest.raises(Exception):
+        test_map = Map(dataset=dataset, segmentation=boundary_segmentation)
+        test_map = Map()
+
+        test_map = Map(dataset=boundary_segmentation)
+        test_map = Map(dataset=5)
+
+        test_map = Map(segmentation=dataset)
+        test_map = Map(segmentation=5)
+
+
+def test_load_overlay(dataset_thingstaette, boundary_segmentation):
```

Comment:
Is the THingstaette data set actually in the repo? I cannot find it right now (independently of our LFS problems)

Reply:
apparently it is not, but if we decide to not use the Westport dataset, we should probably include it.

Reply:
The dataset is not published under an open license, we should be looking for a replacement.